src/workerd/api/sockets.c++

```diff
@@ -238,14 +252,9 @@ jsg::Ref<Socket> connectImplNoOutputLock(
 jsg::Ref<Socket> connectImpl(
     jsg::Lock& js, kj::Maybe<jsg::Ref<Fetcher>> fetcher, AnySocketAddress address,
     jsg::Optional<SocketOptions> options) {
-  jsg::Ref<Fetcher> actualFetcher = nullptr;
-  KJ_IF_MAYBE(f, fetcher) {
-    actualFetcher = kj::mv(*f);
-  } else {
-    actualFetcher = jsg::alloc<Fetcher>(
-        IoContext::NULL_CLIENT_CHANNEL, Fetcher::RequiresHostAndProtocol::YES);
-  }
-  return connectImplNoOutputLock(js, kj::mv(actualFetcher), kj::mv(address), kj::mv(options));
+  // TODO(soon): Doesn't this need to check for the presence of an output lock, and if it finds one
+  // then wait on it, before calling into connectImplNoOutputLock?
```

Comment:
good question, iirc I copied `fetch` here https://github.com/cloudflare/workerd/blob/40509ea403cbd077e76bd7e59582e4aa912a37e5/src/workerd/api/http.c%2B%2B#L1644

Reply:
Yes, this should be waiting for the output lock.

Also, more annoyingly, each write to the socket needs to wait.

Reply:
ahh, I must have missed that. What is the purpose of the output lock here?

Reply:
In a Durable Object, after the application has written anything to disk, we block output to the network until those writes have been confirmed durably stored. This ensures that in the case of a sudden machine failure that causes some writes to be lost, there's no possibility that anything else in the world believed those writes had completed successfully. "Normal" databases would not actually report success from the write operation itself until it is durable, but as an optimization, Durable Objects allow the application to keep running in parallel with confirming the write. For this to be safe, it's important that all outputs are properly blocked.

This is described in more detail in this blog post: https://blog.cloudflare.com/durable-objects-easy-fast-correct-choose-three/